pkg/costattribution/active_tracker.go

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package costattribution
+
+import (
+	"bytes"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	"go.uber.org/atomic"
+)
+
+type ActiveSeriesTracker struct {
+	userID                         string
+	labels                         []string
+	maxCardinality                 int
+	activeSeriesPerUserAttribution *prometheus.Desc
+	overflowLabels                 []string
+	observed                       map[string]*atomic.Int64
+	observedMtx                    sync.RWMutex
+	overflowSince                  atomic.Int64
+	overflowCounter                atomic.Int64
+	cooldownDuration               time.Duration
+	logger                         log.Logger
+}
+
+func newActiveSeriesTracker(userID string, trackedLabels []string, limit int, cooldownDuration time.Duration, logger log.Logger) *ActiveSeriesTracker {
+	orderedLables := slices.Clone(trackedLabels)
+	slices.Sort(orderedLables)
+
```

Comment:
We can remove this, because the constructor isn't exported and the manager already clones and sorts the labels before calling this.

Reply:
```suggestion
```

Reply:
addressed in 5c26a0b4dcf3040586b33dafe91fa403c8be0245, moved the order to another caller.